hack/image/sdk/Dockerfile

```diff
@@ -0,0 +1,3 @@
+FROM registry.access.redhat.com/ubi8/ubi
+
+COPY operator-sdk-dev-linux-gnu  /usr/local/bin/operator-sdk
```

Comment:
WDYT about making this a multi-stage dockerfile where stage 1 copies in the project and builds it, and stage 2 copies the binary from stage 1?

That would eliminate the need for the `hack/image/build-sdk-image.sh` script, and would be a good exemplar to convert the other image build scripts and Dockefiles over.

Reply:
Actually it _does_ look like a script may still be necessary so that we can centralize the logic around building the image and then loading it into kind. We could have a generic script (e.g. ./build-image.sh` that takes a context directory, Dockerfile, and image name) that does this work.

Given that, a follow-on probably makes more sense to do a standalone refactor of the existing Dockerfiles, build scripts, and Makefile targets.

Reply:
WDYT about also setting:
```Dockerfile
ENTRYPOINT ["/usr/local/bin/operator-sdk"]
```

Reply:
@joelanford would that not print the help text to dev/null and exit immediately?